packages/payload/src/collections/operations/updateByID.ts

```diff
@@ -123,11 +125,39 @@ export const updateByIDOperation = async <
     // Retrieve document
     // /////////////////////////////////////
 
+    const where = { id: { equals: id } }
+
+    let fullWhere = combineQueries(where, accessResults)
+
+    const isSoftDeleteAttempt =
+      collectionConfig.softDeletes &&
+      typeof data === 'object' &&
+      data !== null &&
+      'deletedAt' in data &&
+      data.deletedAt != null &&
+      !overrideAccess
```

Comment:
I think `!overrideAccess` should be moved out of this variable creation and used alongside the check below. If it's left like this the variable is a little overloaded.

Same goes for other places this var is created.